backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -174,6 +177,76 @@ export const InitEmailTemplateMiddleware = (
     }
   }
 
+  /**
+   * TODO: Refactor and cleanup
+   * Set selected list and populate the EmailMessages table
+   * @param req
+   * @param res
+   * @param next
+   */
+  const selectListHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<Response | void> => {
+    const CHANNEL_TYPE = ChannelType.Email
+    const userId = req.session?.user?.id
+    const { campaignId } = req.params
+
+    const { list_id: listId } = req.body
+    const logMeta = { campaignId, action: 'selectListHandler' }
+
+    try {
+      const list = await ListService.getList({
+        listId,
+        userId,
+        channel: CHANNEL_TYPE,
+      })
+      if (!list) throw new Error('Error: List not found')
+
+      const { s3key: s3Key, etag, filename } = list
+
+      // check if template exists
+      const template = await EmailTemplateService.getFilledTemplate(+campaignId)
+      if (template === null) {
+        throw new Error(
+          'Error: No message template found. Please create a message template before uploading a recipient file.'
+        )
+      }
+
+      // Store temp filename
+      await UploadService.storeS3TempFilename(+campaignId, filename)
+
+      // Enqueue upload job to be processed
+      await EmailTemplateService.enqueueUpload({
+        campaignId: +campaignId,
+        template,
+        s3Key,
+        etag,
+        filename,
+      })
+
+      // Return early because bulk insert is slow
```

Comment:
I think this comment is a bit misleading cos our processing happened elsewhere (as suggested from the `enqueueUpload` function call while this one might imply that it happens in this same place

Reply:
Resolved [9dad668](https://github.com/opengovsg/postmangovsg/pull/1816/commits/9dad668792a789f19da3dcf90e36e61e8644c843)